app/models/song.rb

```diff
@@ -16,15 +16,35 @@ class Song < ActiveRecord::Base
   validates :chord_sheet, presence: true
   validates_inclusion_of :key, in: VALID_KEYS, allow_nil: true
   validates_inclusion_of :tempo, in: VALID_TEMPOS, allow_nil: true
-  before_save :normalize
+  before_save :normalize, :extract_lyrics
 
   def to_s
     name
   end
 
   private
+  # Titlize important fields and strip away unnecessary spaces
   def normalize
-    self.name = self.name.titleize
-    self.artist = self.artist.titleize if self.artist
+    self.name = self.name.titleize.strip
+    self.artist = self.artist.titleize.strip if self.artist
+    
+    normalized_lines = []
+    self.chord_sheet.split("\n").each { |line| normalized_lines << line.rstrip }
+    self.chord_sheet = normalized_lines.join("\n")
+  end
+
+  def extract_lyrics
+    lyrics = self.chord_sheet.split("\n").find_all { |line| is_line_of_lyrics(line) }
+    self.lyrics = lyrics.join("\n")
+  end
+
+  CHORD_GIVEAWAY_STRINGS = Set.new(%w(# / [ 2 3 4 7 9))
+  def is_line_of_lyrics(line)
+    # headers always end with :
+    return false if line[-1] == ":"
+    # chord lines contain these giveaway substrings
+    return false if line.split.any? {|word| CHORD_GIVEAWAY_STRINGS.include? word}
+    # chord lines have a high density of spaces or contain less than 3 words
+    return (line.count(" ") / line.length.to_f < 0.31) && (line.length > 6)
```

Comment:
Would this work for something like this:
```
C/E   F      C/E     F        Gm  F      C/E   Dm
One name holds weight a  -  bove them all
C/E  F       C/E   F     Gm   F     C/E  Dm
His   fame out - lasts the   earth He  formed
F/A  Bb      F/A   Bb  C  Bb   F/A  Dm
His   praise resounds beyond the  stars
```
This is a pretty extreme case but just want to make sure the density isn't too loose. 

Also, I'm pretty sure there's a number of songs with 3 or less word length lines. No other name again has a nice edge case:

```
     F
Jesus
```